ir/instr.cpp

```diff
@@ -567,7 +567,8 @@ expr BinOp::getTypeConstraints(const Function &f) const {
                   lhs->getType() == rhs->getType();
 
     if (auto ty = getType().getAsStructType()) {
-      instrconstr &= ty->numElements() == 2 &&
+      auto cnt = ty->numElements();
+      instrconstr &= cnt - expr::mkInt(ty->numPaddingsConst(), cnt) == 2 &&
```

Comment:
can you create a function in aggregateType for this, e.g. numElementsExcludingPadding().
Seems useful in general.

Reply:
Updated